### DISCUSSION TITLE
mnt Ordner enthält kein boot/config.txt

### DISCUSSION THREAD

**Discussion by KarlKaracho** [OpenCCU/OpenCCU#1709] [🇩🇪 Fragen&Antworten] [answered]
### Describe the issue you are experiencing

Ich habe Home Assistant OS 7.2 installiert und das funktioniert soweit ich weiß auch wunderbar.
Habe mir das Pi HM-MOD-RPI-PCB Funkmodul gekauft und auch erfolgreich eingebaut.
Nur leider kann ich das Booten des GPIO Moduls nicht aktivieren, da mein mnt Ordner leer ist.

### Describe the behavior you expected

Was mache ich da falsch?

### Steps to reproduce the issue

Habe schon gelesen, dass es am Port liegen kann. Habe Port 22222 und 22 ausprobiert. Bei beiden ist der Ordner leer.

Wenn ich einfach einen Boot Ordner anlege und die config.txt erstelle und da den angegebenen Text hinein schreibe, geht das zwar. Nur verschwindet beides wieder, wenn ich den Host neu starte :-)

### What is the version this bug report is based on?

Home Assistant OS 7.2

### Which base platform are you running?

rpi4 (RaspberryPi4)

### Which HomeMatic/homematicIP radio module are you using?

HM-MOD-RPI-PCB

### Anything in the logs that might be useful for us?

```txt
Mounting /data as /usr/local (Home Assistant Add-On): OK
Starting watchdog...
Identifying onboard hardware: oci, OK
Initializing RTC Clock: no hardware found
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Starting logging: OK
Init onboard LEDs: init, OK
Starting irqbalance: OK
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.4, OK
Identifying Homematic RF-Hardware: HmRF: none, HmIP: none, OK
Updating Homematic RF-Hardware: no GPIO/USB connected RF-hardware found
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting ha-proxy: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: not required
Starting rfd: no BidCos-RF hardware found
Starting HMIPServer: .......OK
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: disabled
Starting Third-Party Addons: OK
Starting crond: OK
Setup onboard LEDs: booted, OK
```

### Additional information

Das ist das Protokoll von Terminal&SSH

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] cli: applying... 
[fix-attrs.d] cli: exited 0.
[fix-attrs.d] profile: applying... 
[fix-attrs.d] profile: exited 0.
[fix-attrs.d] ssh: applying... 
[fix-attrs.d] ssh: exited 0.
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] apks.sh: executing... 
[cont-init.d] apks.sh: exited 0.
[cont-init.d] keygen.sh: executing... 
[22:09:08] INFO: Restoring host keys...
[cont-init.d] keygen.sh: exited 0.
[cont-init.d] profile.sh: executing... 
[cont-init.d] profile.sh: exited 0.
[cont-init.d] ssh.sh: executing... 
[22:09:09] INFO: Setup password login
[cont-init.d] ssh.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[22:09:10] INFO: Starting Web Terminal...
[2022/02/07 22:09:10:1829] N: ttyd 1.6.3-3e37e33 (libwebsockets 4.2.1-v4.2.1)
[2022/02/07 22:09:10:1830] N: tty configuration:
[2022/02/07 22:09:10:1830] N:   start command: tmux -u new -A -s homeassistant bash -l
[2022/02/07 22:09:10:1831] N:   close signal: SIGHUP (1)
[2022/02/07 22:09:10:1831] N:   terminal type: xterm-256color
[2022/02/07 22:09:10:1831] N: LWS: 4.2.1-v4.2.1, loglevel 7
[2022/02/07 22:09:10:1831] N: NET CLI SRV H1 H2 WS ConMon IPV6-off
[2022/02/07 22:09:10:1985] N:    /usr/lib/libwebsockets-evlib_uv.so
[2022/02/07 22:09:10:2409] N:  Using foreign event loop...
[2022/02/07 22:09:10:2409] N:  ++ [wsi|0|pipe] (1)
[2022/02/07 22:09:10:2410] N:  ++ [vh|0|netlink] (1)
[2022/02/07 22:09:10:2412] N:  ++ [vh|1|default||8099] (2)
[2022/02/07 22:09:10:2412] N: lws_socket_bind: nowsi: source ads 0.0.0.0
[2022/02/07 22:09:10:2413] N:  ++ [wsi|1|listen|default||8099] (2)
[2022/02/07 22:09:10:2413] N:  Listening on port: 8099
[22:09:10] INFO: Starting the SSH daemon...
Server listening on 0.0.0.0 port 22.
Server listening on :: port 22.
Accepted password for root from 192.168.178.54 port 61521 ssh2
Accepted password for root from 192.168.178.54 port 61656 ssh2

**Comment by jens-maus** [comment, marked as answer]
Tut mir leid, aber du hast die Anleitung nicht richtig studiert. Du musst natürlich auf dem Host das /mnt/boot suchen und dort die config.txt editieren und nicht innerhalb des Addon Containers. Geh also wie in der Anleitung vor und nutze das Port 22222 addon oder schliesse Tastatur + Keyboard an und geb dort dann "login" ein um in die shell Umgebung des Hosts zu gelangen und dort die config.txt zu editiern.

Ein Bug ist das aber definitiv nicht!

**Comment by KarlKaracho** [comment]
Hi,
danke für deine schnelle Rückmeldung.
Dass das ein Bug ist will ich ja auch gar nicht sagen.

Eher das ich zu doof bin :-)

Ok, habe das mit dem Port 22222 Addon gerade auch noch einmal ausprobiert. Das gibt mir folgendes Protokoll aus:

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
no vda1 config found
no sda1 config found
no sdb1 config found
no mmcblk0p1 config found
no mmcblk1p1 config found
no nvme0n1p1 config found
This Configurator did it's job. Perform a hard-power-off now. This configurator only works once and is no longer needed.


Wenn ich mich nach dem "hard-power-off" Neustart wieder entweder mit Putty und Port 22222 einwähle, oder mit dem Home Assistant Terminal. Sind an beiden Stellen der mnt Ordner leer.

Weil so wie ich das das 22222 Addon verstehe muss ich da ja nichts machen, außer zu installieren und zu starten. Weil in den Einstellungen stand der "Key" schon drin.

Kannst du mir bitte noch weiter helfen.

Danke dir.

**Reply by KarlKaracho** [reply]
Habe auch schon probiert die SD Karte am PC zu bearbeiten. Nur da erscheinen leider gar nicht erst die Dateien. Es kommt eine hassos-boot Partition und ein Laufwerk, auf das kann ich aber nicht zugreifen.

Auf der hassos-boot Partition sind allerhand Dateien, aber keine config.txt oder ähnliches.

**Reply by jens-maus** [reply]
> Wenn ich mich nach dem "hard-power-off" Neustart wieder entweder mit Putty und Port 22222 einwähle, oder mit dem Home Assistant Terminal. Sind an beiden Stellen der mnt Ordner leer.
> 
> Weil so wie ich das das 22222 Addon verstehe muss ich da ja nichts machen, außer zu installieren und zu starten. Weil in den Einstellungen stand der "Key" schon drin.

Das hört sich für mich immer noch an, das du das 22222 Add-on nicht richtig verwendest. Lies dazu mal genau die Dokumentation durch damit du verstehst für was dieser "Key" ist und was du da selbst eintragen musst und wie du dich auf Port 22222 dann verbinden kannst.

Oder du schließst eben Tastatur+Monitor an denen HA Host an und logsst dich dort ein und gibst `login` dort ein um in die Konsole des Hosts (und nicht eines Containers) zu gelangen und dann muss es da ein `/mnt/boot` Verzeichnis geben das nicht leer sein wird und wo bereits eine `config.txt` Datei zu finden sein muss (wenn das ein RaspberryPi ist den du verwendest). Und dann musst du entsprechend der RaspberryMatic Anleitung vorgehen.

**Comment by KarlKaracho** [comment]
Hi Jens,
danke für die Antwort.
Ja genau ich habe mich falsch eingeloggt. Danach habe ich noch eine andere Anleitung gefunden mit der ich auch schließlich die Datei überhaupt gefunden habe.

Ich schildere hier mal kurz was ich gemacht habe, falls es noch so einen "naja nicht IT Profi wie mich gibt :-)"

- das alles ist los gelöst vom SSH Termin in Home Assistant, da war ich länger auf dem falschen Weg
- zuerst wie in der Anleitung erklärt einen Key für SSH mit Port 22222 erstellt, das habe ich mit Puttygen gemacht
- diesen auf einen USB Stick kopiert (wie in der Anleitung beschrieben)
- den dann per USB importiert (geht beides gut "Power-Off-Reset" oder der Import über Home Assistant / Host/ ... import per USB)
- danach kann man sich per Putty auf Port 22222 einloggen
   - hier war bei mir wichtig, den privat Key zu laden (unter Connection / SSH / Auth)
   - macht man das nicht kommt man auch nicht rein / bzw wird gesperrt :-)
- ist man eingeloggt editiert man die config.txt mit dem Befehl: "vi/mnt/boot/config.txt"
- die Taste "i" drücken um in den Edit Mode zu kommen
- so kann man die Datei editieren und einfach den Text aus der Anleitung einfügen
- Danach "esc" drücken (Edit Mode verlassen)
- Danach ":" und dies mit "x" bestätigen zum speichern
- anschließend den Host neu starten
- danach würde bei mir das "HM-MOD-RPI-PCB" erkannt

**Comment by KarlKaracho** [comment]
Jetzt habe ich aber direkt eine neue Frage.
Das HM-MOD-RPI-PCB Modul wir im Addon erkannt, und wird auch in der WEB UI von RaspberryMatic unter homematicIP-RF (HmIP) Info und HomeMatic-RF (HmRF) Info angezeigt.
Nur kann ich keine Geräte anlernen.

Da bekomme ich die Meldung:
Es wurde keinerlei nutzbares Homematic Funkmodul/LAN - Gateway für die homematic oder homematicIP Kommuniaktion im System gefunden.

Meines erachtens habe ich die Anleitung auch durchgeschaut, aber da finde ich nichts, oder bin... wieder... zu doof.

Ich habe diesen Beitrag gefunden:
https://github.com/jens-maus/RaspberryMatic/issues/1448
Ist das das gleiche Problem?

Kann mir da bitte einer weiterhelfen.

**Reply by jens-maus** [reply]
> Jetzt habe ich aber direkt eine neue Frage.

Neue Frage = Neuer Beitrag. Das ist hier kein Chatkanal, folglich solltest du für jede neue thematische Frage auch einen neuen Beitrag eröffnen.

**Comment by KarlKaracho** [comment]
Hat keiner eine Antwort dazu?

Ich habe auch noch diesen Beitrag gefunden:
https://homematic-forum.de/forum/viewtopic.php?t=69982&p=682561

Habe auch mal bei ELV gefragt, die meinen ich könnte das Modul gerne einschicken zum prüfen... hmm...

Haltet ihr das für realistisch, dass das Modul kaputt ist, obwohl es richtig angezeigt wird?
